schema/meter_provider.json

```diff
@@ -6,14 +6,20 @@
     "additionalProperties": false,
     "properties": {
         "readers": {
-            "type": "object",
-            "additionalProperties": true,
-            "patternProperties": {
-                "^periodic.*": {
-                    "$ref": "#/$defs/PeriodicMetricReader"
-                },
-                "^pull.*": {
-                    "$ref": "#/$defs/PullMetricReader"
+            "type": "array",
+            "items": {
+                "title": "MetricReader",
+                "type": "object",
+                "additionalProperties": false,
+                "minProperties": 1,
+                "maxProperties": 1,
+                "properties": {
+                    "periodic": {
+                        "$ref": "#/$defs/PeriodicMetricReader"
+                    },
+                    "pull": {
+                        "$ref": "#/$defs/PullMetricReader"
+                    }
```

Comment:
is a patternProperties missing for the readers definition?

Reply:
I don't think so. Every reader should either be a pull reader with an exporter specified as an argument, or a periodic reader (aka push) with an exporter defined as an argument. The way we've structured it, I don't believe there are any additional "types" or readers.